tests/test_engine.py

```diff
@@ -263,19 +271,54 @@ def _assert_scraped_items(self):
                 self.assertEqual('Item 2 name', item['name'])
                 self.assertEqual('200', item['price'])
 
-    def _assert_signals_catched(self):
-        assert signals.engine_started in self.run.signals_catched
-        assert signals.engine_stopped in self.run.signals_catched
-        assert signals.spider_opened in self.run.signals_catched
-        assert signals.spider_idle in self.run.signals_catched
-        assert signals.spider_closed in self.run.signals_catched
+    def _assert_bytes_received(self):
```

Comment:
Could you please add a test which shows the signal behavior when the data is larger than a buffer, i.e. when a signal is sent several times by Scrapy?

Reply:
Added in 7753e6bed0aabc73b336dcdcf314f6b4e7be9855.
Not sure about the `2**14` thing though, it doesn't seem to slow the test down but there might be a better way to force a smaller buffer size.

Running only `tests/test_engine.py` with the large payload:
```
real	0m4.765s
user	0m2.445s
sys	0m0.330s
```

without it:
```
real	0m4.751s
user	0m2.367s
sys	0m0.338s
```